xml/src/main/java/org/ehcache/xml/ss/ReflectionHelper.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.xml.ss;
+
+import org.ehcache.spi.service.Service;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+class ReflectionHelper {
+
+  static Service instantiate(Class<? extends Service> clazz, List<String> unparsedArgs) throws Exception {
```

Comment:
We're already using a repackaged version of `ConstructorUtils` in the class instance provider code for precisely this purpose. You may be able to reuse that logic.

Reply:
I would need to modify `ConstructorUtils` to add the parameter type conversion logic. I'll have a look to figure out the best way to merge both these util classes.